tests/sys/vane/mesa/mesa-pact.hoon

```diff
@@ -75,18 +73,39 @@
   =^  moves-12  ames.bud  (ames-reply:v ames.bud ~[/pact] moves-11 bex-roof) :: "ok, here is the 5th fragment"
   =^  moves-13  ames.nec  (ames-reply:v ames.nec ~[/pact] moves-12 bex-roof) :: "ok, here is the complete message"
   (ames-expect-msg:v moves-13 dat)
-:: ++  test-dire-ns
-::   =/  bex-roof  (make-roof //some/data/atom atom+!>((bex (bex 14))))
-::   =/  res
-::     %-  scry:(ames.nec ~1111.1.10 `@`0xdead.beef bex-roof)
-::     =-  [~ / %x [[~nec %$ ud+1] -]]
-::     %+  weld
-::       /mess/0/pact/13/pure/init
-::     /publ/[(scot %ud nec-life)]/g/x/0/dap//some/data/atom
-::   ?~  res
-::     ~&  %sig  ~
-::   ?~  u.res
-::     ~&  %inner-sig  ~
-::   ~&  >  test-dire-ns/;;([@tas pact:pact:ames.nec] [p q.q]:u.u.res)
-::   ~
+++  test-mesa-ns
+  =/  bex-roof  (make-roof //some/data/atom atom+!>((bex (bex 14))))
+  =/  res
+    %-  scry:(ames.nec ~1111.1.10 `@`0xdead.beef bex-roof)
+    =-  [~ / %x [[~nec %$ ud+1] -]]
+    %+  weld
+      /mess/0/pact/13/pure/init
+    /publ/[(scot %ud nec-life)]/g/x/0/dap//some/data/atom
+  ?~  res
+    ~&  %sig  ~
+  ?~  u.res
+    ~&  %inner-sig  ~
+  ~&  >  test-mesa-ns/;;([@tas pact:pact:ames.nec] [p q.q]:u.u.res)
+  ~
+::
+++  test-mesa-verify-msg-fail
+  ~?  >  dbug  'test-mesa-verify-msg-fail'
+  ::  4-fragment message
+  ::
+  =/  dat          (crip (reap 3.500 'A'))
```

Comment:
@lukechampine this is the test that fails to verify in `+verify-msg:verifier` here: `?>  =(?=(^ pair) (expect-pair state))`.

Extended output of the fragment, proof, and state of the verifier:

https://gist.github.com/yosoyubik/3a8cbb16b251044787cfc6377d62ec81

Reply:
yeah I think it's safe to just remove `expect-pair`. Its definition is changing with the new traversal code, and removing it won't break anything in the interim.

Reply:
Then we can't make progress from the second fragment; verify-msg just returns `~`, i think this is the same behavior we saw doing real tests between two ships.

This is what we pass into the verifier:

```
>>> [ %no-state
    los
  [ leaves=4
    leaf=2
    cur=[l=2 r=4]
    leaf-queue=~
    parent-stack=~[0xb121.29ae.2118.9091.fac2.fe02.b665.a550.34a6.b38a.8616.767d.58bb.b9ff.b4f0.f4c4]
  ]
    dat
  42.774.436.690.957.487.429.920.950.773.873.837.743.069.182.823.381.674.213.013.050.438.812.601.248.985.070.400.637.813.118.635.372.047.998.325.168.476.746.089.141.563.561.575.575.845.631.803.128.042.257.050.944.270.835.248.713.
450.570.141.561.773.648.949.221.948.896.141.709.476.007.716.544.570.298.404.884.710.638.845.251.054.364.745.800.009.903.364.702.635.294.512.925.491.236.699.482.388.443.051.211.017.739.235.358.046.881.398.192.588.443.622.139.608.4
25.389.256.286.021.617.484.075.643.872.657.595.932.523.297.570.808.530.027.229.409.623.791.287.260.424.113.749.651.212.559.076.019.787.373.071.627.198.660.237.146.761.073.821.286.754.711.039.912.059.575.094.087.308.460.538.134.59
9.112.473.530.495.211.099.812.019.678.147.517.684.145.065.431.265.014.889.562.813.217.646.937.292.592.961.398.237.259.433.775.352.607.549.621.259.048.979.083.511.648.344.561.627.675.718.761.969.477.057.659.780.240.730.794.181.427
.824.163.451.193.240.134.744.953.291.046.723.379.654.408.093.933.751.584.505.323.115.091.485.389.339.898.333.151.803.667.132.108.092.547.858.435.961.640.141.833.312.102.783.131.647.304.455.375.792.420.212.084.590.846.083.814.942.
236.169.055.982.937.657.912.123.941.995.227.632.643.570.563.316.421.761.606.561.685.586.634.696.604.995.442.755.782.291.687.001.057.951.566.316.577.082.807.075.870.228.527.817.254.586.722.127.166.329.827.416.369.649.433.587.598.5
78.679.001.082.798.535.632.485.622.298.868.940.926.981.210.923.046.099.817.129.016.301.734.683.684.535.933.924.317.145.628.841.970.810.600.470.750.224.529.248.412.092.448.613.875.644.552.042.391.315.708.976.979.133.621.677.649.21
3.161.629.846.697.876.281.292.318.273.448.136.465.785.401.512.157.184.475.237.605.098.226.578.381.480.857.012.294.529.170.218.896.533.280.478.886.508.247.386.525.723.208.674.106.432.857.597.106.646.333.283.435.437.396.717.602.202
.934.212.275.822.958.383.738.197.049.075.271.262.078.912.983.459.205.070.726.955.796.257.818.513.538.380.534.226.524.076.285.916.462.990.526.366.868.568.924.354.494.877.939.201.171.247.379.951.751.482.971.692.277.623.261.925.221.
319.542.774.970.140.463.512.274.627.232.266.304.115.884.756.424.247.389.414.888.184.048.320.751.324.153.648.453.537.888.514.329.529.102.755.927.670.896.984.624.999.116.346.726.602.470.106.223.507.849.099.263.994.611.537.375.909.5
62.311.555.178.876.323.008.009.177.522.396.194.468.821.615.572.042.217.497.514.619.462.530.841.240.029.667.033.373.290.432.304.855.746.648.344.165.046.156.023.126.743.777.659.172.286.457.457.624.948.434.445.123.198.758.311.550.19
8.095.587.894.560.738.602.637.852.493.937.447.226.410.036.420.547.040.204.987.632.055.073.249.206.063.155.857.922.343.844.140.259.054.158.355.971.400.302.664.400.861.846.681.781.905.327.219.431.063.376.464.805.447.214.692.380.546
.822.473.964.350.219.488.759.245.785.488.821.114.297.968.066.900.539.394.568.523.757.953.475.444.442.702.208.196.246.588.914.565.678.771.135.791.032.687.372.911.088.562.779.335.671.141.910.736.526.356.920.007.788.852.838.700.199.
878.278.007.418.840.836.845.432.345.994.805.533.851.476.031.124.532.064.719.229.920.393.409.020.525.405.037.855.629.303.460.558.118.745.693.713.320.016.424.430.431.404.526.965.123.617.814.516.219.437.289.232.700.044.985.224.087.3
43.788.447.199.285.547.807.482.610.841.919.630.888.705.772.019.830.018.655.521.290
  pair=~
]
```

That pair=~ looks suspicious...